### PR TITLE
Handle negative long values in TempData correctly

### PR DESCRIPTION
**Pull request by ajaybhargavb**
Issue - #3524 
@pranavkm @kichalla

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/SessionStateTempDataProvider.cs [hunk 1]
```diff
@@ -132,7 +132,7 @@ public class SessionStateTempDataProvider : ITempDataProvider
                     else if (item.Value is long)
                     {
                         var longValue = (long)item.Value;
-                        if (longValue < int.MaxValue)
+                        if (longValue >= int.MinValue && longValue < int.MaxValue)
```
Comment (kichalla) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/SessionStateTempDataProvider.cs]: `longValue <= int.MaxValue`?

Comment (ajaybhargavb) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/ViewFeatures/SessionStateTempDataProvider.cs]: :+1:

**Comment by pranavkm** [comment]
Could you also add a test for this? Might just have to add test data for existing scenarios.

:watch:

**Comment by ajaybhargavb** [comment]
Updated.

**Comment by kichalla** [comment]
:shipit:

File: test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewFeatures/SessionStateTempDataProviderTest.cs [hunk 2]
```diff
@@ -170,14 +170,17 @@ public void SaveAndLoad_StringCanBeStoredAndLoaded()
             Assert.Equal("value", stringVal);
         }
 
-        [Fact]
-        public void SaveAndLoad_IntCanBeStoredAndLoaded()
+        [Theory]
+        [InlineData(10)]
+        [InlineData(int.MaxValue)]
+        [InlineData(int.MinValue)]
```
Comment (pranavkm) [inline test/Microsoft.AspNet.Mvc.ViewFeatures.Test/ViewFeatures/SessionStateTempDataProviderTest.cs]: :100:

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**